### DISCUSSION TITLE
Delete flow should have access to deleted data

### DISCUSSION THREAD

**Discussion by FINDarkside** [directus/directus#15660] [Archive]
Right now a delete flow only gets the id of deleted item, which isn't very useful since the item has already been deleted and you can't read it anymore.

Related: https://github.com/directus/directus/discussions/2794 and https://github.com/directus/v8-archive/issues/842

**Comment by br41nslug** [comment]
If you use `Filter (blocking)` instead of `Action (non-blocking)` that runs before the item is deleted so it's possible to retrieve data from the object that will be deleted

**Reply by copostic** [reply]
I just tried with a `Filter (blocking)` flow but I still just have the id : 

`{
   '$trigger': {
     payload: [ '28' ],
     event: 'instance.items.delete',
     collection: 'instance'
   },
   '$last': {
     payload: [ '28' ],
     event: 'instance.items.delete',
     collection: 'instance'
   },
   '$accountability': {
     user: '925907c1-d9c0-4b34-b018-236c88ab8453',
     role: '8509887b-cb41-4f6f-b246-e318ea10d739',
     admin: true,
     app: true,
     ip: 'XX.XX.XX.XX,
     userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36',
     origin: 'https:/my-url/',
     permissions: []
   },
   '$env': {
     TOTO: 'TESTING',
     TEST: 'europe-west1-b'
   }
 }
`

**Reply by br41nslug** [reply]
Filter also just gets the id indeed as thats the only value used for a delete action, however since a filter is triggered before the actual delete of data you can still fetch the fields you want from the database while in an action this data doesnt exist anymore.
